src/service.py

```diff
@@ -4,11 +4,16 @@
 from logging import getLogger
 from typing import Any, Optional
 
+import yaml
 from charms.operator_libs_linux.v2 import snap
+from ops.model import Model, ModelError
+from PySquashfsImage import SquashFsImage
```

Comment:
instead of using a 3rd party library I would prefer to use snap tooling for security reasons, even if we have to use subprocess:
```
$ snap download golang-openstack-exporter
Fetching snap "golang-openstack-exporter"
Fetching assertions for "golang-openstack-exporter"
Install the snap with:
   snap ack golang-openstack-exporter_97.assert
   snap install golang-openstack-exporter_97.snap

$ snap info ./golang-openstack-exporter_97.snap                        
path:       "golang-openstack-exporter_97.snap"
name:       golang-openstack-exporter
summary:    OpenStack Exporter for Prometheus
version:    v1.4.0 -
build-date: 2021-05-26
license:    unset
description: |
  The OpenStack exporter, exports Prometheus metrics from a running OpenStack
  cloud for consumption by prometheus. The cloud credentials and identity
  configuration should use the os-client-config format and must by specified
  with the --os-client-config flag.
  
  Other options as the binding address/port can by explored with the --help
  flag.
  
  By default the openstack_exporter serves on port 0.0.0.0:9180 at the /metrics
  URL.
commands:
  - golang-openstack-exporter.openstack-exporter
```

Reply:
+1 to no extra dependencies; shelling out to `snap` is a nice solution :ok_hand:

Reply:
My initial implementation was using the subprocess and parsing the output, but I think this solution is very fragile. Any change in the output of the cli tool and this crashes the solution. They don't have any commitment on keeping the output as it is.

If we are going down that path I think I prefer using subprocess using `squashfs` and then reading the files

Reply:
Closing this topic once we are not reading the name from the resource file anymore